plugins/aggregators/starlark/starlark.go

```diff
@@ -0,0 +1,137 @@
+package starlark //nolint
+
+import (
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+	common "github.com/influxdata/telegraf/plugins/common/starlark"
+	"go.starlark.net/starlark"
+)
+
+const (
+	description  = "Aggregate metrics using a Starlark script"
+	sampleConfig = `
+  ## The Starlark source can be set as a string in this configuration file, or
+  ## by referencing a file containing the script.  Only one source or script
+  ## should be set at once.
+  ##
+  ## Source of the Starlark script.
+  source = '''
+def add(cache, metric):
+  cache["last"] = metric
+
+def push(cache, accumulator):
+  last = cache.get("last")
+  if last != None:
+	  accumulator.add_fields(last.name, last.fields, last.tags)
+'''
```

Comment:
should the push automatically clear the cache or should that be in the example?

Reply:
The cache is automatically cleared [on Reset](https://github.com/influxdata/telegraf/pull/9419/files#diff-e85b90e00ee23e36285bc26a1d203912e20d681bb6435d13b3a408d9d7a6e781R116)